pkg/resource/license.go

```diff
@@ -0,0 +1,119 @@
+// you may not use this file except in compliance with the Elastic License.
+
+package resource
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	commonlicense "github.com/elastic/cloud-on-k8s/pkg/controller/common/license"
+	"github.com/elastic/cloud-on-k8s/pkg/controller/common/version"
+	"github.com/elastic/cloud-on-k8s/pkg/controller/license"
+	"github.com/elastic/cloud-on-k8s/pkg/utils/k8s"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	// defaultOperatorLevel is the default license level when no operator license is installed
+	defaultOperatorLevel = "basic"
+	// minSupportedVersion is the minimum Elastic stack version used to find an operator license
+	minSupportedVersion = "6.8.0"
+	// licensingCfgMapName is the name of the config map used to store licensing information
+	licensingCfgMapName = "elastic-licensing"
+)
+
+// LicensingInfo represents information about the operator license including the total memory of all Elastic managed
+// components
+type LicensingInfo struct {
+	Timestamp               string `json:"timestamp"`
+	LicenseLevel            string `json:"license_level"`
+	MemoryInGiga            string `json:"memory_in_giga"`
+	EnterpriseResourceUnits string `json:"enterprise_resource_units"`
+}
+
+// LicensingResolver resolves the licensing information of the operator
+type LicensingResolver struct {
+	operatorNs string
+	client     k8s.Client
+}
+
+// ToInfo returns licensing information given the total memory of all Elastic managed components
+func (r LicensingResolver) ToInfo(totalMemory resource.Quantity) LicensingInfo {
+	eru := inEnterpriseResourceUnits(totalMemory)
+	memoryInGiga := inGiga(totalMemory)
+	licenseLevel := r.getOperatorLicenseLevel()
+
+	return LicensingInfo{
+		Timestamp:               time.Now().Format(time.RFC3339),
+		LicenseLevel:            licenseLevel,
+		MemoryInGiga:            memoryInGiga,
+		EnterpriseResourceUnits: eru,
+	}
+}
+
+// Save updates or creates licensing information in a config map
+func (r LicensingResolver) Save(info LicensingInfo, operatorNs string) error {
+	data, err := info.toMap()
+	if err != nil {
+		return err
+	}
+
+	log.V(1).Info("Saving", "license_info", info)
+	cm := corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: operatorNs,
+			Name:      licensingCfgMapName,
+		},
+		Data: data,
+	}
+	err = r.client.Update(&cm)
+	if apierrors.IsNotFound(err) {
+		return r.client.Create(&cm)
+	}
+	return err
+}
+
+// getOperatorLicenseLevel gets the level of the operator license.
+// If no license is found, the defaultOperatorLevel is returned.
+func (r LicensingResolver) getOperatorLicenseLevel() string {
+	checker := commonlicense.NewLicenseChecker(r.client, r.operatorNs)
+	minVersion, _ := version.Parse(minSupportedVersion)
+	lic, _, found := license.FindLicense(r.client, checker, minVersion)
```

Comment:
This is incorrect as `findLicense` returns a stack level license not an orchestration level license. 

I would suggest to refactor 
https://github.com/elastic/cloud-on-k8s/blob/571a78aaf8e2dbc49d839e12c48c15dde7ccb3f2/pkg/controller/common/license/check.go#L49

to something along those lines (Pseudo-code!)
```go
// EnterpriseFeaturesEnabled returns true if a valid enterprise license is installed.
func (lc *checker) CurrentLIcense() (*EnterpriseLicense, error) {
	licenses, err := EnterpriseLicenses(lc.k8sClient)
	if err != nil {
		return nil, errors.Wrap(err, "failed to list enterprise licenses")
	}

       sort.Slice(licenses, func(i, j int) bool {
		t1, t2 := EnterpriseLicenseTypeOrder[licenses[i].Type],
			 EnterpriseLicenseTypeOrder[licenses[j].Type
		if t1 != t2 { // sort by type
			return t1 < t2
		}
		// and by remaining time
		return licenses[i].remaining < licenses[j].remaining
	})
	for _, l := range licenses {
		valid, err := lc.Valid(l)
		if err != nil {
			return nil, err
		}
		if valid {
			return l, nil
		}
	}
	return nil, nil
}

// EnterpriseFeaturesEnabled returns true if a valid enterprise license is installed.
func (lc *checker) EnterpriseFeaturesEnabled() (bool, error) {
	license, err := lc.CurrentLicense(lc.k8sClient) // or similar name
	if err != nil {
		return false, err
	}
       return license != nil 
}

```